package.json

```diff
@@ -34,22 +34,23 @@
     "async-iterators": "^0.2.2",
     "eslint": "^3.12.2",
     "eslint-config-loopback": "^7.0.1",
-    "mocha": "^2.1.0",
-    "should": "^8.0.2"
+    "mocha": "^3.2.0",
+    "should": "^11.1.2"
   },
   "dependencies": {
-    "async": "~1.0.0",
+    "async": "~2.1.4",
     "bluebird": "^3.1.1",
     "debug": "^2.1.1",
     "depd": "^1.0.0",
     "inflection": "^1.6.0",
-    "loopback-connector": "^2.1.0",
+    "loopback-connector": "^3.0.0",
     "minimatch": "^3.0.3",
-    "node-uuid": "^1.4.2",
-    "qs": "^3.1.0",
+    "qs": "^6.3.0",
```

Comment:
AFAICT, `qs` is used only to parse dataSource/connector `settings` when a single URL string is provided.

qs's changelog can be found here: https://github.com/ljharb/qs/blob/8bd4c6cf12898f469838980317fec92007e5112a/CHANGELOG.md

Here is an overview of breaking changes:

 - 6.0 is just dropping support for Node versions before 4
 - 5.0 reverts back `allowDots` to false. This flag was introduced in 3.x defaulting to true (see https://github.com/ljharb/qs/commit/33888bb8eb3bc96adaf0dc0dc3f9d61c55d252a2). 
 - 4.0 reverts back handling of keys that overwrite built-in Object properties to the behaviour of 2.x versions, see https://github.com/ljharb/qs/pull/98. IMO, typical connector settings shouldn't be using built-in property names, thus we shouldn't be affected by this change.

As I see it, the only thing that may break after this update is connection strings using `key.subkey=value` notation instead of `key[subkey]=value`. Considering that the dot notation is not documented in LoopBack docs, I think upgrade to `qs@4+` shouldn't cause any significant breakage of existing apps.